include/aws/http/private/http_impl.h

```diff
@@ -64,6 +64,9 @@ enum aws_http_header_name {
     AWS_HTTP_HEADER_RETRY_AFTER,
     AWS_HTTP_HEADER_VARY,
     AWS_HTTP_HEADER_WARNING,
+    AWS_HTTP_HEADER_UPGRADE,
+    AWS_HTTP_HEADER_KEEP_ALIVE,
+    AWS_HTTP_HEADER_PROXY_CONNECTION,
```

Comment:
Add a todo to our transformation function:

> An intermediary transforming an HTTP/1.x message to HTTP/2 MUST remove connection-specific header fields as discussed in [Section 7.6.1](https://rfc-editor.org/rfc/rfc9110#section-7.6.1) of [[HTTP](https://httpwg.org/http2-spec/draft-ietf-httpbis-http2bis.html#RFC9110)], or their messages will be treated by other HTTP/2 endpoints as [malformed](https://httpwg.org/http2-spec/draft-ietf-httpbis-http2bis.html#malformed)

Reply:
there already has a TODO for it. I guess I can just do it in this PR

```
/* TODO: handle connection-specific header field (RFC7540 8.1.2.2) */
```